### PR TITLE
[clang-tidy] Fix Hungarian Prefix in readability-identifier-naming

### LINKED ISSUE #80268: [Bug] Clang-tidy: Suggests removing and duplicating Hungarian Notation prefixes at the same time
`Clang-tidy` suggests removing and duplicating the Hungarian Notation prefixes at the same time. 

Here is a minimum reproducible example using `clang-tidy-18`:

* `Clang-tidy` version:
```bash
$ clang-tidy-18 --version
Ubuntu LLVM version 18.1.0
  Optimized build.
```

* `.clang-tidy` config file:
<summary>
<details>

```yaml
---
FormatStyle: file
WarningsAsErrors: '*'
Checks: >
  -*,
  readability-identifier-naming
CheckOptions:
  readability-identifier-naming.GlobalVariablePrefix: g
  readability-identifier-naming.GlobalPointerPrefix: g
  readability-identifier-naming.TypedefSuffix: _t
  readability-identifier-naming.UnionSuffix: _t
  readability-identifier-naming.TypeAliasSuffix: _t

  readability-identifier-naming.ClassCase: CamelCase
  readability-identifier-naming.VariableCase: CamelCase
  readability-identifier-naming.ParameterCase: CamelCase
  readability-identifier-naming.PointerParameterCase: CamelCase
  readability-identifier-naming.ConstantParameterCase: CamelCase
  readability-identifier-naming.ConstantPointerParameterCase: CamelCase

  readability-identifier-naming.HungarianNotation.PrimitiveType.void: v
  readability-identifier-naming.HungarianNotation.PrimitiveType.int8_t: s8
  readability-identifier-naming.HungarianNotation.PrimitiveType.int16_t: s16
  readability-identifier-naming.HungarianNotation.PrimitiveType.int32_t: s32
  readability-identifier-naming.HungarianNotation.PrimitiveType.int64_t: s64
  readability-identifier-naming.HungarianNotation.PrimitiveType.float: f32
  readability-identifier-naming.HungarianNotation.PrimitiveType.double: f64
  readability-identifier-naming.HungarianNotation.UserDefinedType.PVOID: t
  readability-identifier-naming.HungarianNotation.UserDefinedType.INT8: t
  readability-identifier-naming.HungarianNotation.UserDefinedType.INT16: t
  readability-identifier-naming.HungarianNotation.UserDefinedType.INT32: t
  readability-identifier-naming.HungarianNotation.UserDefinedType.INT64: t
  readability-identifier-naming.HungarianNotation.UserDefinedType.uint8_t: t
  readability-identifier-naming.HungarianNotation.UserDefinedType.UINT16: t
  readability-identifier-naming.HungarianNotation.UserDefinedType.uint32_t: t
  readability-identifier-naming.HungarianNotation.UserDefinedType.UINT64: t
  readability-identifier-naming.HungarianNotation.UserDefinedType.float: t
  readability-identifier-naming.HungarianNotation.UserDefinedType.double: t

  readability-identifier-naming.HungarianNotation.General.TreatStructAsClass: true
  readability-identifier-naming.HungarianNotation.DerivedType.Array: r
  readability-identifier-naming.HungarianNotation.DerivedType.Pointer: p
  readability-identifier-naming.HungarianNotation.DerivedType.FunctionPointer: p

  readability-identifier-naming.VariableHungarianPrefix: LowerCase
  readability-identifier-naming.ClassHungarianPrefix: LowerCase
  readability-identifier-naming.LocalPointerHungarianPrefix: LowerCase
  readability-identifier-naming.GlobalPointerHungarianPrefix: LowerCase
  readability-identifier-naming.ParameterHungarianPrefix: LowerCase
  readability-identifier-naming.PointerParameterHungarianPrefix: LowerCase
  readability-identifier-naming.ConstantParameterHungarianPrefix: LowerCase
  readability-identifier-naming.ConstantPointerParameterHungarianPrefix: LowerCase
  readability-identifier-naming.TypedefHungarianPrefix: LowerCase
  readability-identifier-naming.TypeAliasHungarianPrefix: LowerCase
  readability-identifier-naming.UnionHungarianPrefix: LowerCase
  readability-identifier-naming.GlobalVariableHungarianPrefix: LowerCase
```

</details>
</summary>

* `test.c`:

```c
#include <stdint.h>

typedef uint32_t MyType32_t;
typedef uint8_t MyType8_t;

void my_func(
    MyType32_t t_MyType32,
    MyType32_t *pt_MyType32,
    MyType8_t t_MyType8,
    MyType8_t *pt_MyType8,
    const MyType32_t t_ConstMyType32,
    const MyType32_t *pt_ConstMyType32,
    const MyType8_t t_ConstMyType8,
    const MyType8_t *pt_ConstMyType8,
    uint32_t u32_Uint32,
    uint32_t *pu32_Uint32,
    uint8_t u8_Uint8,
    uint8_t *pu8_Uint8,
    const uint32_t u32_ConstUint32,
    const uint32_t *pu32_ConstUint32,
    const uint8_t u8_ConstUint8,
    const uint8_t *pu8_ConstUint8
)
{}
```

* Run command:
```bash
$ clang-tidy-18 test.c
```

* Expected result:
`clang-tidy` finds no errors

* Actual result:

<summary>
<details>

```bash
$ clang-tidy-18 test.c 
Error while trying to load a compilation database:
Could not auto-detect compilation database for file "test.c"
No compilation database found in /path/to/my/file or any parent directory
fixed-compilation-database: Error while opening fixed database: No such file or directory
json-compilation-database: Error while opening JSON database: No such file or directory
Running without flags.
16 warnings generated.
error: invalid identifier naming option 'TypeAliasHungarianPrefix' [clang-tidy-config,-warnings-as-errors]
error: invalid identifier naming option 'TypedefHungarianPrefix' [clang-tidy-config,-warnings-as-errors]
/path/to/my/file/test.c:8:16: error: invalid case style for parameter 't_MyType32' [readability-identifier-naming,-warnings-as-errors]
    8 |     MyType32_t t_MyType32,
      |                ^~~~~~~~~~
      |                MyType32
/path/to/my/file/test.c:9:17: error: invalid case style for pointer parameter 'pt_MyType32' [readability-identifier-naming,-warnings-as-errors]
    9 |     MyType32_t *pt_MyType32,
      |                 ^~~~~~~~~~~
      |                 p_PtMyType32
/path/to/my/file/test.c:10:15: error: invalid case style for parameter 't_MyType8' [readability-identifier-naming,-warnings-as-errors]
   10 |     MyType8_t t_MyType8,
      |               ^~~~~~~~~
      |               MyType8
/path/to/my/file/test.c:11:16: error: invalid case style for pointer parameter 'pt_MyType8' [readability-identifier-naming,-warnings-as-errors]
   11 |     MyType8_t *pt_MyType8,
      |                ^~~~~~~~~~
      |                p_PtMyType8
/path/to/my/file/test.c:12:22: error: invalid case style for constant parameter 't_ConstMyType32' [readability-identifier-naming,-warnings-as-errors]
   12 |     const MyType32_t t_ConstMyType32,
      |                      ^~~~~~~~~~~~~~~
      |                      ConstMyType32
/path/to/my/file/test.c:13:23: error: invalid case style for pointer parameter 'pt_ConstMyType32' [readability-identifier-naming,-warnings-as-errors]
   13 |     const MyType32_t *pt_ConstMyType32,
      |                       ^~~~~~~~~~~~~~~~
      |                       p_PtConstMyType32
/path/to/my/file/test.c:14:21: error: invalid case style for constant parameter 't_ConstMyType8' [readability-identifier-naming,-warnings-as-errors]
   14 |     const MyType8_t t_ConstMyType8,
      |                     ^~~~~~~~~~~~~~
      |                     ConstMyType8
/path/to/my/file/test.c:15:22: error: invalid case style for pointer parameter 'pt_ConstMyType8' [readability-identifier-naming,-warnings-as-errors]
   15 |     const MyType8_t *pt_ConstMyType8,
      |                      ^~~~~~~~~~~~~~~
      |                      p_PtConstMyType8
/path/to/my/file/test.c:17:15: error: invalid case style for pointer parameter 'pu32_Uint32' [readability-identifier-naming,-warnings-as-errors]
   17 |     uint32_t *pu32_Uint32,
      |               ^~~~~~~~~~~
      |               pu32_Pu32Uint32
/path/to/my/file/test.c:19:14: error: invalid case style for pointer parameter 'pu8_Uint8' [readability-identifier-naming,-warnings-as-errors]
   19 |     uint8_t *pu8_Uint8,
      |              ^~~~~~~~~
      |              pu8_Pu8Uint8
/path/to/my/file/test.c:21:21: error: invalid case style for pointer parameter 'pu32_ConstUint32' [readability-identifier-naming,-warnings-as-errors]
   21 |     const uint32_t *pu32_ConstUint32,
      |                     ^~~~~~~~~~~~~~~~
      |                     pu32_Pu32ConstUint32
/path/to/my/file/test.c:23:20: error: invalid case style for pointer parameter 'pu8_ConstUint8' [readability-identifier-naming,-warnings-as-errors]
   23 |     const uint8_t *pu8_ConstUint8
      |                    ^~~~~~~~~~~~~~
      |                    pu8_Pu8ConstUint8
14 warnings treated as errors
``` 

</details>
</summary>

So it seems `clang-tidy` suggests removing the Hungarian notation prefixes from any `typedef` types, but at the same time wants to duplicate the prefixes of primitive types. 

I commented on #64864, but since I don't quite have the same issue, I decided to raise a new ticket with more details.

### PR DESCRIPTION
**Pull request by PiotrZSL**
Fix handling of Hungarian Prefix when configured to LowerCase.
  Warnings no longer will be emited for names that already match
  this style.

Fixes: #80268

### CODE REVIEW

File: clang-tools-extra/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -229,7 +229,8 @@ Changes in existing checks
 
 - Improved :doc:`readability-identifier-naming
   <clang-tidy/checks/readability/identifier-naming>` check in `GetConfigPerFile`
-  mode by resolving symbolic links to header files.
+  mode by resolving symbolic links to header files. Fixed handling of Hungarian
```
Comment (dmpolukhin) [inline clang-tools-extra/docs/ReleaseNotes.rst:232]: I think it should be separate item in Release notes unrelated to symlinks.

**Comment by dmpolukhin** [comment]
I'm not an expert in Hungarian naming but from what I know changes looks good to me. Just nit in release notes.

File: clang-tools-extra/docs/ReleaseNotes.rst (hunk 1 above)
Comment (PiotrZSL) [reply clang-tools-extra/docs/ReleaseNotes.rst:232]: One entry per check. One sentence per change.

Comment (dmpolukhin) [reply clang-tools-extra/docs/ReleaseNotes.rst:232]: > One entry per check. One sentence per change.

I don't think it is common practice, see https://releases.llvm.org/17.0.1/tools/clang/tools/extra/docs/ReleaseNotes.html I see 6 entries about `readability-identifier-naming`. Does it change recently and documented somewhere?

Comment (PiotrZSL) [reply clang-tools-extra/docs/ReleaseNotes.rst:232]: It's not documented, just an "idea". Release notes in Clang-tidy 17 were a little bit unreadable, so in Calng-tidy 18 I push a little bit to unify them: https://releases.llvm.org/18.1.0/tools/clang/tools/extra/docs/ReleaseNotes.html

Comment (dmpolukhin) [reply clang-tools-extra/docs/ReleaseNotes.rst:232]: OK, if you think it makes Release Notes more readable.

**Review by dmpolukhin — APPROVED** [review]

**Pull request merged**